rts/motoko-rts/src/bitmap.rs

```diff
@@ -0,0 +1,116 @@
+use crate::alloc::alloc_blob;
+use crate::mem::memzero;
+use crate::types::{size_of, Blob, Bytes, Obj};
+
+/// Current bitmap
+static mut BITMAP_PTR: *mut u8 = core::ptr::null_mut();
+
+pub unsafe fn alloc_bitmap(heap_size: Bytes<u32>) {
+    // We will have at most this many objects in the heap, each requiring a bit
+    let n_bits = heap_size.to_words().0;
+    // Each byte will hold 8 bits.
+    let bitmap_bytes = (n_bits + 7) / 8;
+    // Also round allocation up to 8-bytes to make iteration efficient. We want to be able to read
+    // 64 bits in a single read and check as many bits as possible with a single `word != 0`.
+    let bitmap_bytes = Bytes(((bitmap_bytes + 7) / 8) * 8);
+    // Allocating an actual object here as otherwise dump_heap gets confused
+    let blob = alloc_blob(bitmap_bytes).unskew() as *mut Blob;
+    memzero(blob.payload_addr() as usize, bitmap_bytes.to_words());
+
+    BITMAP_PTR = blob.payload_addr()
+}
+
+pub unsafe fn free_bitmap() {
+    BITMAP_PTR = core::ptr::null_mut();
+}
+
+pub unsafe fn get_bit(idx: u32) -> bool {
+    let byte_idx = idx / 8;
+    let byte = *BITMAP_PTR.add(byte_idx as usize);
+    let bit_idx = idx % 8;
+    (byte >> bit_idx) & 0b1 == 0b1
```

Comment:
Would it be worth checking this generates wasm bittest instruction? Might be faster, but who knows. I think there a bittest intrinsic in Rust now. Or just make an issue for later.

Reply:
Which instruction do you mean? I don't see a bittest instruction here: https://webassembly.github.io/spec/core/appendix/index-instructions.html

Reply:
If there even are dedicated bittest instructions - I couldn't find any. Maybe I was thinking of x86

Reply:
Just confirmed with Andreas that there aren't any. Don't bother.